src/loop.c

```diff
@@ -0,0 +1,69 @@
+// this is a default router firmware supporting only wormhole routing and rmap on port 0
+
+#include <stdint.h>
+
+#include "platform.h"
+
+//#define DEBUG
+
+inline int32_t get_path (uint32_t taddr);
+
+void loop () {
+    int i;
+    uint32_t taddr;
+    int32_t found_address;
+
+    // setup
+    *table_size_addr = (uint32_t)&table_size;
+    *table_addr = (uint32_t)table;
+
+    while (enable) {
+        for (i=0;i<*port_num;i++) {
+            if (GOT_PKG(i)) {
+                taddr = TADDR(i);
+                found_address = get_path(taddr);
+#ifdef DEBUG
+                printf ("got pkg from port %d taddr=%d, tport=%d\n", i, taddr, found_address);
```

Comment:
Suggest making a logging function which is only defined if `debug` is defined. Will make the code cleaner.